core/src/main/java/com/graphhopper/GHRequest.java

```diff
@@ -40,6 +38,10 @@
     private boolean possibleToAdd = false;
     private Locale locale = Locale.US;
 
+    // List of preferred start (1st element) and arrival directions (all other). 
+    // Directions are north based azimuth (clockwise) in (0,2pi) or NaN for equal preference
+    private List<Double> preferredDirections;
```

Comment:
no, I do not mean this. But I can refactor once you are done and you can review and see how this feels.